openstack/clientconfig/requests.go

```diff
@@ -155,8 +179,33 @@ func GetCloudFromYAML(opts *ClientOpts) (*Cloud, error) {
 		if !ok {
 			return nil, fmt.Errorf("cloud %s does not exist in clouds-public.yaml", profileName)
 		}
+		cloud, err = mergeClouds(cloud, publicCloud)
```

Comment:
If this merge correct? Doesn't this makes the `clouds-public.yaml` to override `clouds.yaml`, should be the other way around.

`secure.yaml` overrides `clouds.yaml` which overrides values coming from `clouds-public.yaml` if `profile` is defined.

Reply:
I think it should be correct. The order of arguments is `mergeClouds(override, cloud interface{})`

If you can think of better variable names in the mergeClouds function let me know

Reply:
You're right, I got it backwards there. 

Sometimes the hardest part is to find good names. 😄 I have to think if there are some better names.